sdk/typescript/src/utils/ed25519-hd-key.ts

```diff
@@ -69,26 +67,11 @@ export const getPublicKey = (
   return withZeroByte ? newArr : signPk;
 };
 
-export const isValidPath = (path: string): boolean => {
-  if (!pathRegex.test(path)) {
-    return false;
-  }
-  return !path
-    .split('/')
-    .slice(1)
-    .map(replaceDerive)
-    .some(isNaN as any /* ts T_T*/);
-};
-
 export const derivePath = (
-  path: Path,
+  path: HardenedEd25519Path,
   seed: Hex,
   offset = HARDENED_OFFSET,
 ): Keys => {
-  if (!isValidPath(path)) {
```

Comment:
I'd keep this. Not everyone uses TypeScript

Reply:
Good catch... forgot there are folks that might use plain JS sans type declaration files. I'll revise this!

Reply:
This method is only used internally and in one place (`Ed25519Keypair.deriveKeypair`) which does the initial runtime validation check, so I think it should be fine to nix this validation. LMK if you think otherwise though and glad to add a version of this back :)